{
  "Severity": "MODERATE",
  "UpdatedAt": "2022-06-16T23:55:32Z",
  "Package": {
    "Ecosystem": "RUST",
    "Name": "futures-util"
  },
  "Advisory": {
    "DatabaseId": 160001,
    "Id": "GSA_kwCzR0hTQS01cjlnLWo3amotaHc2Y84AAnEB",
    "GhsaId": "GHSA-5r9g-j7jj-hw6c",
    "References": [
      {
        "Url": "https://nvd.nist.gov/vuln/detail/CVE-2020-35908"
      },
      {
        "Url": "https://rustsec.org/advisories/RUSTSEC-2020-0062.html"
      },
      {
        "Url": "https://github.com/rust-lang/futures-rs/issues/2050"
      },
      {
        "Url": "https://github.com/advisories/GHSA-5r9g-j7jj-hw6c"
      }
    ],
    "Identifiers": [
      {
        "Type": "GHSA",
        "Value": "GHSA-5r9g-j7jj-hw6c"
      },
      {
        "Type": "CVE",
        "Value": "CVE-2020-35908"
      }
    ],
    "Description": "An issue was discovered in the futures-util crate before 0.3.2 for Rust. FuturesUnordered can lead to data corruption because Sync is mishandled.",
    "Origin": "UNSPECIFIED",
    "PublishedAt": "2022-05-24T17:37:49Z",
    "Severity": "MODERATE",
    "Summary": "Improper `Sync` implementation on `FuturesUnordered` in futures-utils can cause data corruption",
    "UpdatedAt": "2023-01-27T05:02:59Z",
    "WithdrawnAt": "",
    "CVSS": {
      "Score": 5.5,
      "VectorString": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
    }
  },
  "Versions": [
    {
      "FirstPatchedVersion": {
        "Identifier": "0.3.2"
      },
      "VulnerableVersionRange": "\u003c 0.3.2"
    }
  ]
}